students/JerryH/Lesson04/trigrams.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import string
+import random
+from collections import defaultdict
+
+
+def get_text():
+    file = open("sherlock.txt", "r")
+    text = file.read()
+    file.close()
+    return text
+
+
+def get_word_list():
+    text = get_text().replace('\n', ' ').lower()
+    punc = string.punctuation.replace("-", "") #keep "-" in the punctuation
+    table = str.maketrans({key: None for key in punc})
+    new_text = text.translate(table)
+
+    word = new_text.split()
+    return word
+
+# create a default list dictionary
+word_dict = defaultdict(list, {})
+words = get_word_list()
+
+new_wordings = [] # Beginning of the new wordings
+r = random.randint(0, len(words))
+
+# Create trigrams in dictionary 'word_dict'
+for i in range(len(words)-2):
+    word_dict[words[i] + ' ' + words[i+1]].append(words[i+2])
```

Comment:
you should be able to move above code into `main()` 

it's common to either put this code under `__name__ == '__main__'` block or into it's own function. basically you either write it as a script or you write it with function/classes